service/lib/agama/software/manager.rb

```diff
@@ -420,9 +421,66 @@ def import_gpg_keys
       end
 
       def add_base_repos
+        # support multiple labels/installation media?
+        label = product.labels.first
+
+        if label
+          logger.info "Installation repository label: #{label.inspect}"
+          # we cannot use the simple /dev/disk/by-label/* device file as there
+          # might be multiple devices with the same label
+          device = installation_device(label)
+          if device
+            logger.info "Installation device: #{device}"
+            repositories.add("hd:/?device=" + device)
+            return
+          end
+        end
+
+        # disk label not found or not configured, use the online repositories
         product.repositories.each { |url| repositories.add(url) }
       end
 
+      # find all devices with the required disk label
+      # @return [Array<String>] returns list of devices, e.g. `["/dev/sr1"]`,
+      # returns empty list if there is no device with the required label
+      def disks_with_label(label)
+        data = list_disks
+        disks = data.fetch("blockdevices", []).map do |device|
+          device["kname"] if device["label"] == label
+        end
+        disks.compact!
+        logger.info "Disks with the installation label: #{disks.inspect}"
+        disks
+      end
+
+      # get list of disks, returns parsed data from the `lsblk` call
+      # @return [Hash] parsed data
+      def list_disks
+        # we need only the kernel device name and the label
+        output = `lsblk --paths --json --output kname,label`
```

Comment:
`lsblk` is included in `util-linux-systemd`. Should we add a dependency?

Reply:
I'll add that...